### PR TITLE
Add button variants to the Link component

### PR DESCRIPTION
**Pull request by djalmaaraujo**
## Description
image

Sometimes, we want to display links as Buttons, and this is a way to reuse Button styles within the `Link` component.

## Checklist

- [x] This PR has good automated test coverage
- [x] The storybook for the component has been updated

## Steps to Test

1. Pull down PR.
2. `npm run dev`.
3. Open http://localhost:6006/?path=/docs/navigation-link--docs
4. There's a new section called: Button Variants
5. You should see a demo for each of the following new variants: 

```
button-primary
button-secondary
button-tertiary
button-danger
button-display
button-ghost
```

### CODE REVIEW

File: src/system/Button/Button.tsx [hunk 1]
```diff
@@ -46,16 +46,6 @@ const Button = forwardRef< HTMLButtonElement, ButtonProps >(
...
-					verticalAlign: 'middle',
-					display: 'inline-flex',
-					alignItems: 'center',
-					justifyContent: 'center',
-					minHeight: '36px',
-					py: 0,
-					textDecoration: 'none',
-					'&:hover': {
-						textDecoration: 'none',
-					},
```
Comment (andrea-sdl) [inline src/system/Button/Button.tsx:58]: @djalmaaraujo why this change?

File: src/system/theme/index.js [hunk 2]
```diff
@@ -492,7 +523,7 @@ export default {
 			fontFamily: 'body',
 			lineHeight: 'body',
 			fontWeight: 'body',
-			fontSize: 2,
+			fontSize: '16px',
```
Comment (andrea-sdl) [inline src/system/theme/index.js]: @djalmaaraujo I see some visual differences if I compare trunk with this branch

Let's thake the accordion
**Trunk**
https://vip-design-system-components.netlify.app/?path=/story/accordion--default
CleanShot 2024-04-18 at 16 45 28@2x


**This branch** 
https://deploy-preview-374--vip-design-system-components.netlify.app/?path=/story/accordion--default
CleanShot 2024-04-18 at 16 45 41@2x

Also, links and buttons have bigger fonts. Is this intentional?

**Review by brunobasto — APPROVED** [review]
Looks great 🚀
